tracer/src/Datadog.Trace/Tagging/TraceSources.cs

```diff
@@ -0,0 +1,31 @@
+// <copyright file="TraceSources.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.SourceGenerators;
+
+namespace Datadog.Trace.Tagging
+{
+    /// <summary>
+    /// Trace source products (to be used in the '_dd.p.ts' tag)
+    /// </summary>
+    [Flags]
+    [EnumExtensions]
+    internal enum TraceSources : byte
+    {
+        None = 0,
+        APM = 1,
+        ASM = 2,
+        DSM = 4,
+        DJM = 8,
+        DBM = 16,
```

Comment:
nit picking, but .NET naming conventions call for `Apm`, `Asm`, `Dsm`, etc on 3-letter acronyms (see also `Xml`, `Sql`).

Reply:
Also, I like to use this pattern because it's easier to see the number sequence (0, 1, 2, 3, ...) instead of having to do the math in my head 😅 
```csharp
Apm = 1 << 0,
Asm = 1 << 1,
Dsm = 1 << 2,
```

definitely not a blocker, though

Reply:
Fine. On it.